lib/src/socket_client.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:uuid/uuid.dart';
+
+import '../graphql_flutter.dart';
+
+SocketClient socketClient;
+
+class SocketClient {
+  static Future<SocketClient> connect(final String endPoint, {
+    final List<String> protocols = const ['graphql-ws'],
+    final Map<String, String> headers = const {'Content-Type': 'application/json'}
+  }) async {
+    return SocketClient(GraphQLSocket(await WebSocket.connect(
+        endPoint, protocols: protocols, headers: headers)));
```

Comment:
comma after `headers: headers`

Reply:
Are you sure we want to enforce trailing comma for one-line declarations?
I understand the motivation for all commas on multi-line assignments, but the syntax definitely seems odd for when it is adjacent to a brace. Maybe we'd have to follow this up with an *always chop* or *closing brace on new line*

eg) 
```dart
return SocketClient(GraphQLSocket(await WebSocket.connect(
  endPoint, protocols: protocols, headers: headers,
)));
```
or
```dart
return SocketClient(GraphQLSocket(await WebSocket.connect(
  endPoint, 
  protocols: protocols, 
  headers: headers,
)));
```

Reply:
Ah, flutter actually backs me up this time 😃 

> Use a trailing comma for arguments, parameters, and list items, but only if they each have their own line.

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-a-trailing-comma-for-arguments-parameters-and-list-items-but-only-if-they-each-have-their-own-line

Reply:
@cal-pratt We use format on save, which automatically puts the code on multiple lines if there is a tailing comma. That's why we can always enforce the tailing comma, and still comply with the style guide ;)

Besides that, is's the preferred way to write widgets: [https://flutter.io/formatting/#using-trailing-commas](https://flutter.io/formatting/#using-trailing-commas).

Reply:
Cool. I'm going to take the time and go through the flutter spec to clean up this code. 
Should have a new revision up sometime this weekend